packages/langium/src/workspace/workspace-lock.ts

```diff
@@ -59,8 +64,25 @@ export class DefaultWorkspaceLock implements WorkspaceLock {
         return this.enqueue(this.writeQueue, action, tokenSource.token);
     }
 
-    read<T>(action: () => MaybePromise<T>): Promise<T> {
-        return this.enqueue(this.readQueue, action);
+    read<T>(action: () => MaybePromise<T>, priority?: boolean): Promise<T> {
```

Comment:
Just thinking loud: Is this the first time we have a "priority" queue?
You could refactor a common component/function if not.

Reply:
The workspace lock is a mutex, not a queue. It features some very specific semantics about read/read with prio/write actions that are likely not relevant for a common component.